### DISCUSSION TITLE
Add KeyDown KeyUp and KeyPress Events !

### DISCUSSION THREAD

**Discussion by maiad** [dotnet/maui#17710] [Ideas]
There is no reason why universal applications should not support keyboard input. It's absurd to write an application where I can't handle keyboard input. And I don't mean your Entry !

**Comment by jonmdev** [comment]
This works well for at least mouse input from my testing, and likely would work well for keyboard too:

https://github.com/TolikPylypchuk/SharpHook

**Reply by maiad** [reply]
I tried, and apart from MAUI forcing me to take such a ridiculous detour, I had trouble converting Keys to characters. SharpHook was returning blank characters and such. Maybe it was my fault, but it doesn't change the fact that this is what MAUI should have made available to me.

**Reply by maiad** [reply]
Not to mention that I consider adding a keyboard listener to MAUI a trivial matter.

**Comment by MathijsWouters** [comment]
Yeah i'm struggling with this too

**Comment by pme442** [comment]
I am also having difficulty with this.

**Comment by inescastro05** [comment]
Could anyone solve it for android?

**Reply by stephenquan** [reply]
@inescastro05 are you connecting a keyboard to your Android device?

**Reply by inescastro05** [reply]
es una tablet, no usa teclado fisico
